tools/plugins/com.liferay.ide.core/src/com/liferay/ide/core/ILiferayProject.java

```diff
@@ -14,13 +14,15 @@
 
 package com.liferay.ide.core;
 
+import com.liferay.ide.core.event.EventListener;
+
 import org.eclipse.core.resources.IFolder;
 import org.eclipse.core.resources.IProject;
 
 /**
  * @author Gregory Amerson
  */
-public interface ILiferayProject {
+public interface ILiferayProject extends EventListener {
```

Comment:
Not forced to implement and the `BaseLiferayProject` have a empty implement. if not, those listener registry code would't carry out.  If we try to register each `GradleProject` class to a `ResourceChangeListener` in pre logic, and we are developing 50 modules, then we have 50 `ResourceChangeListener` registered, This may slow down the performance.